typedefs.h

```diff
@@ -38,6 +39,16 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #endif
 #endif
 
+// OpenStreetMap node ids are higher than 2^32
+OSRM_STRONG_TYPEDEF(uint64_t, NodeID64)
+OSRM_STRONG_TYPEDEF(uint32_t, OSMWayID)
+
+static const NodeID64 INVALID_NODEID64 = NodeID64(std::numeric_limits<uint64_t>::max());
+static const OSMWayID INVALID_OSM_WAYID = OSMWayID(std::numeric_limits<uint32_t>::max());
+
+using NodeID64_weak = uint64_t;
+using EdgeID64_weak = uint64_t;
+
```

Comment:
If possible, set the fixed integer types once and then use type inference, so that in case you change the types, the code will adapt itself, like:

    std::numeric_limits<NodeID64>::max()

    std::numeric_limits<decltype(node)>::max()